feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts

```diff
@@ -243,4 +255,44 @@ export class ConfiguratorGroupMenuComponent {
   isConflictGroupType(groupType: Configurator.GroupType): boolean {
     return this.configuratorGroupsService.isConflictGroupType(groupType);
   }
+
+  /**
+   * Returns group-status style classes dependent on completeness, conflicts, visited status and configurator type.
+   *
+   * @param {Configurator.Group} group - Current group
+   * @param {Configurator.Configuration} configuration - Configuration
+   * @return {Observable<boolean>} - true if visited and not a conflict group
+   */
+  getGroupStatusStyles(
+    group: Configurator.Group,
+    configuration: Configurator.Configuration
+  ): Observable<string> {
+    return this.isGroupVisited(group, configuration).pipe(
+      switchMap((isVisited) => {
+        const CLOUDCPQ_CONFIGURATOR_TYPE = 'CLOUDCPQCONFIGURATOR';
+        let groupStatusStyle: string = 'cx-menu-item';
+        if (
+          configuration.owner?.configuratorType !==
+            CLOUDCPQ_CONFIGURATOR_TYPE &&
+          !group.consistent
+        ) {
+          groupStatusStyle = groupStatusStyle + this.WARNING;
+        }
+        if (
+          configuration.owner?.configuratorType !==
+            CLOUDCPQ_CONFIGURATOR_TYPE &&
+          group.complete &&
+          group.consistent &&
+          isVisited
+        ) {
+          groupStatusStyle = groupStatusStyle + this.COMPLETE;
+        }
+        if (!group.complete && isVisited) {
+          groupStatusStyle = groupStatusStyle + this.ERROR;
+        }
+        return of(groupStatusStyle);
+      }),
+      take(1)
```

Comment:
`isGroupVisited` already have `take(1)`, if we use `map`, we can remove this `take(1)`. I even think we can remove or deprecate that `isGroupVisited` function.

Reply:
picked

Reply:
Agree for the un-needed take(1), but I would rather leave the isGroupVisited method. It has a value on its own, could be used in views that render the group information differently.